sympy/integrals/prde.py

```diff
@@ -188,11 +188,29 @@ def prde_linear_constraints(a, b, G, DE):
         N = max([ri.degree(DE.t) for _, ri in Q])
         M = Matrix(N + 1, m, lambda i, j: Q[j][1].nth(i))
     else:
-        M = Matrix()  # No constraints, return the empty matrix.
+        M = Matrix(0, m, [])  # No constraints, return the empty matrix.
 
     qs, _ = list(zip(*Q))
     return (qs, M)
 
+def poly_linear_constraints(p, d):
```

Comment:
Is this function an improvement over the [prde_linear_constraints](https://github.com/sympy/sympy/blob/master/sympy/integrals/prde.py#L166)?